appservice/src/deploy/waitForDeploymentToComplete.ts

```diff
@@ -105,3 +109,28 @@ async function getLatestDeployment(kuduClient: KuduClient, permanentId: string |
 
     return [deployment, permanentId, initialStartTime];
 }
+
+async function waitForDeploymentToStart(kuduClient: KuduClient, id: string): Promise<void> {
```

Comment:
I feel like all of this logic can be more tightly integrated with the existing code rather than being it's own function. For example, `expectedId` seems very similar to `permanentId` and you've got two loops with two polling intervals when you could just have one

Biggest thing you'd have to change is this, but I'm sure you could refactor somehow (since `deployment` would probably be undefined until it actually starts):
```
if (deployment === undefined || !deployment.id) {
        throw new Error(localize('failedToFindDeployment', 'Failed to get status of deployment.'));
}
```